frontend/dockerfile/dockerfile_test.go

```diff
@@ -3463,6 +3464,51 @@ COPY --from=build /dest /dest
 	require.Equal(t, []byte("0644\n0755\n0413\n"), dt)
 }
 
+func testAddGit(t *testing.T, sb integration.Sandbox) {
+	f := getFrontend(t, sb)
+
+	dockerfile := []byte(`
+FROM alpine
+
+# Basic case
+ADD https://github.com/moby/buildkit.git#v0.4.0 /buildkit
```

Comment:
We should avoid network pulls where possible. Other git tests use a local test repository.

Reply:
Do we need to create an HTTPS test server, or can we just allow plain HTTP for 127.0.0.1?

https://github.com/moby/buildkit/pull/2799#discussion_r855151994

Reply:
I'm ok with plain HTTP for now like other tests if there isn't anything else simple. With `go-git` we would possibly need to rewrite all these cases though.

Reply:
Updated PR to use plain HTTP